File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
       
        Arrays.sort(citations);
        
        for (int i = citations.length; i >= 1; i--)
        {
            int index = citations.length - 1;

            while (citations[index] >= i)
            {
                if (index == citations.length - i)
                {
                    return i;
                }
                index--;
            }
        }
        
        return 0;
    }
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
	int answer = 0;

	sort(citations.begin(), citations.end());


	for (int i = citations.size(); i >= 1; i--)
	{
		int index = citations.size() - 1;

		while (citations[index] >= i)
		{
			if (index == citations.size() - i)
			{
				answer = i;
				i = 0;
				break;
			}
			index--;
		}
	}
	return answer;
}